src/vs/workbench/contrib/notebook/browser/notebookEditorWidget.ts

```diff
@@ -1386,13 +1389,15 @@ export class NotebookEditorWidget extends Disposable implements INotebookEditorD
 			this._localStore.add(this._webview.webview.onDidBlur(() => {
 				this._outputFocus.set(false);
 				this._webviewFocused = false;
+				this._webviewHasFocus.set(false);
 
 				this.updateEditorFocus();
 				this.updateCellFocusMode();
 			}));
 
 			this._localStore.add(this._webview.webview.onDidFocus(() => {
 				this._outputFocus.set(true);
+				this._webviewHasFocus.set(true);
```

Comment:
it looks like you could just use `notebookOutputFocused`

Reply:
But then I'd need to add that same context into the listServices class, that would mean we'd have a cyclic ref.
Notebooks depend on listServices today, and I don't think lists classes (files) should depend on notebook stuff..